test/nm_test.dart

```diff
@@ -2163,4 +2174,91 @@ void main() {
 
     await client.close();
   });
+
+  Future<void> verifyActiveConnections(
+      DBusAddress clientAddress, List<String> expectedIds) async {
+    // active connections are cached => use a temp extra client to verify
+    var client = NetworkManagerClient(bus: DBusClient(clientAddress));
+    await client.connect();
+    expect(client.activeConnections.map((c) => c.id), equals(expectedIds));
+    await client.close();
+  }
+
+  test('activate ethernet connection', () async {
+    var server = DBusServer();
+    var clientAddress =
+        await server.listenAddress(DBusAddress.unix(dir: Directory.systemTemp));
+
+    var nm = MockNetworkManagerServer(clientAddress);
+    await nm.start();
+    await nm.addDevice(deviceType: NetworkManagerDeviceType.ethernet.index);
+    var s1 = await nm.addConnectionSettings();
+    var s2 = await nm.addConnectionSettings();
+
+    var client = NetworkManagerClient(bus: DBusClient(clientAddress));
+    await client.connect();
+
+    expect(client.devices, hasLength(1));
+    var device = client.devices[0];
+    expect(client.settings.connections, hasLength(2));
+
+    var connection1 = await client.activateConnection(device: device);
+    expect(connection1.id, equals(s1.path.value));
+    await verifyActiveConnections(clientAddress, [s1.path.value]);
+
+    var connection2 = await client.activateConnection(
+        device: device, connection: client.settings.connections[1]);
+    expect(connection2.id, equals(s2.path.value));
+    await verifyActiveConnections(
+        clientAddress, [s1.path.value, s2.path.value]);
+
+    await client.deactivateConnection(connection1);
+    await verifyActiveConnections(clientAddress, [s2.path.value]);
```

Comment:
We don't need to use a extra client to check these - we can just directly check the server with:
```dart
expect(nm.activeConnections, equals([...]);
```
and
```
expect(nm.activeConnections, isEmpty);
```

Reply:
The problem was that the client's active connection list wouldn't get updated, but I realized that the test server was not informing the client about added and removed active connections. I've changed `MockNetworkManagerServer.add/removeActiveConnection()` to emit a property change signal to make the client aware of the change. Would that be fine?